Microservices/user-service/src/main/java/app/sportahub/userservice/service/user/UserServiceImpl.java

```diff
@@ -68,4 +77,81 @@ public User getUserById(String id) {
         return Optional.ofNullable(userRepository.findUserById(id))
                 .orElseThrow(() -> new UserDoesNotExistException(id));
     }
-}
+
+
+    @Override
+    public Profile updateUserProfile(String id, ProfileRequest profileRequest) {
+        Optional<User> optionalUserById = Optional.ofNullable(userRepository.findUserById(id));
+        if (optionalUserById.isEmpty()) {
+            throw new UserDoesNotExistException(id);
+        }
+        User user = optionalUserById.get();
+        Profile previousProfile = user.getProfile();
```

Comment:
Profile here can be empty.
The response to that request will be

```json
{
    "error": "An unexpected error occurred: Cannot invoke \"app.sportahub.userservice.model.user.Profile.getFirstName()\" because \"previousProfile\" is null",
    "status": 500
}
```